packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,383 @@
+        else if (sel == this.protocolVersionsProxy.selector) protocolVersionsProxy = ProtocolVersions(_address);
+        else revert("DeploySuperchainOutput: Unknown selector");
+    }
+
+    // Save the output to a TOML file.
+    function writeOutputFile(string memory _outfile) public pure {
+        _outfile;
+        require(false, "DeploySuperchainOutput: saveOutput not implemented");
+    }
+
+    function output() public view returns (Output memory) {
+        return Output({
+            superchainProxyAdmin: superchainProxyAdmin,
+            superchainConfigImpl: superchainConfigImpl,
+            superchainConfigProxy: superchainConfigProxy,
+            protocolVersionsImpl: protocolVersionsImpl,
+            protocolVersionsProxy: protocolVersionsProxy
+        });
+    }
+
+    function checkOutput() public view {
+        // Assert that all addresses are non-zero and have code.
+        // We use LibString to avoid the need for adding cheatcodes to this contract.
+        address[] memory addresses = new address[](5);
+        addresses[0] = address(superchainProxyAdmin);
+        addresses[1] = address(superchainConfigImpl);
+        addresses[2] = address(superchainConfigProxy);
+        addresses[3] = address(protocolVersionsImpl);
+        addresses[4] = address(protocolVersionsProxy);
+
+        for (uint256 i = 0; i < addresses.length; i++) {
+            address who = addresses[i];
+            require(who != address(0), string.concat("check failed: zero address at index ", LibString.toString(i)));
+            require(
+                who.code.length > 0, string.concat("check failed: no code at ", LibString.toHexStringChecksummed(who))
+            );
+        }
+
+        // All addresses should be unique.
+        for (uint256 i = 0; i < addresses.length; i++) {
+            for (uint256 j = i + 1; j < addresses.length; j++) {
+                string memory err =
+                    string.concat("check failed: duplicates at ", LibString.toString(i), ",", LibString.toString(j));
+                require(addresses[i] != addresses[j], err);
+            }
+        }
+    }
+}
+
+// For all broadcasts in this script we explicitly specify the deployer as `msg.sender` because for
+// testing we deploy this script from a test contract. If we provide no argument, the foundry
+// default sender is be the broadcaster during test, but the broadcaster needs to be the deployer
+// since they are set to the initial proxy admin owner.
+contract DeploySuperchain is Script {
+    // -------- Core Deployment Methods --------
+
+    // This entrypoint is for end-users to deploy from an input file and write to an output file.
+    // In this usage, we don't need the input and output contract functionality, so we deploy them
+    // here and abstract that architectural detail away from the end user.
+    function run(string memory _infile) public {
```

Comment:
Our other scripts use env var as config - curious if you think using args like this is better?

Reply:
I am open to the input file being an env var, and it's easy to change to that later if we need. I do think this is simpler UX for users since it's just a single `forge script scripts/DeploySuperchain.s.sol "run(string)" myFile.toml` instead of having to set an env var first

Reply:
You could argue that `CONFIG=myfile.toml forge script scripts/DeploySuperchain.s.sol` is more concise but i am not super opinionated here :)

Reply:
Fair, you definitely could argue that

I do like the consistency between each of our `run` methods (the only difference is the input arg you pass), instead of fragmenting UX where some take input args and some take env vars. If you are not super opinionated I would prefer to keep this as-is for now, and we can always change this later easily if we find this inconvenient